common/scala/src/main/scala/whisk/http/PoolingRestClient.scala

```diff
@@ -42,7 +44,13 @@ import spray.json._
  *  on each request because it doesn't need to look up the pool corresponding
  *  to the host. It is also easier to add an extra queueing mechanism.
  */
-class PoolingRestClient(protocol: String, host: String, port: Int, queueSize: Int)(implicit system: ActorSystem) {
+class PoolingRestClient(
+  protocol: String,
+  host: String,
+  port: Int,
+  queueSize: Int,
+  httpFlow: Option[Flow[(HttpRequest, Promise[HttpResponse]), (Try[HttpResponse], Promise[HttpResponse]), Any]] = None)(
```

Comment:
Have you thought about overriding `requestQueue` instead of passing in an arbitrary flow here?